packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -138,15 +142,20 @@ export class ForkChoice implements IForkChoice {
     private readonly fcStore: IForkChoiceStore,
     /** The underlying representation of the block DAG. */
     private readonly protoArray: ProtoArray,
+    validatorCount: number,
     readonly metrics: ForkChoiceMetrics | null,
     private readonly opts?: ForkChoiceOpts,
     private readonly logger?: Logger
   ) {
     this.head = this.updateHead();
     this.balances = this.fcStore.justified.balances;
+    this.voteCurrentIndices = new Array(validatorCount).fill(NULL_VOTE_INDEX);
+    this.voteNextIndices = new Array(validatorCount).fill(NULL_VOTE_INDEX);
+    // when compute deltas, we ignore epoch if voteNextIndex is NULL_VOTE_INDEX anyway
+    this.voteNextEpochs = new Array(validatorCount).fill(INIT_VOTE_EPOCH);
```

Comment:
I don't see how these arrays grow as validator count increases. Need to ensure that all newly created entries are initialized properly (or that all accesses check for uninitialized data).